plugins/modules/sonic_dcbx.py

```diff
@@ -0,0 +1,452 @@
+#
+# Before State:
+# -------------
+#
+# sonic# show running-configuration | grep dcbx
+# sonic#
+
+- name: Modify DCBX configurations
+  dellemc.enterprise_sonic.sonic_dcbx:
+    config:
+      global:
+        enabled: true
+    state: merged
+
+# After State:
+# ------------
+#
+# sonic# show running-configuration | grep dcbx
+# !
+# dcbx enable
+# !
+
+# Using "merged" state for DCBx interface configuration
+#
+# Before State:
+# -------------
+# sonic# show running-configuration interface Ethernet 0
+# !
+# interface Ethernet0
+#  mtu 9100
+#  speed 25000
+#  unreliable-los auto
+#  no shutdown
+#  no dcbx enable
+#  no dcbx tlv-select pfc
+#  no dcbx tlv-select ets-conf
+#  no dcbx tlv-select ets-reco
+
+
+- name: Modify DCBx Interface configurations
+  dellemc.enterprise_sonic.sonic_dcbx:
+    config:
+      interfaces:
+        - name: Ethernet0
+          enabled: true
+          pfc_tlv_enabled: true
+          ets_configuration_tlv_enabled: true
+          ets_recommendation_tlv_enabled: true
+    state: merged
+
+# After State:
+# ------------
+#
+# sonic# show running-configuration interface Ethernet0
+# !
+# interface Ethernet0
+#  mtu 9100
+#  speed 25000
+#  unreliable-los auto
+#  no shutdown
+
+# Using "deleted" state for DCBx interface configuration
+#
+# Before State:
+# -------------
+#
+# sonic# show running-configuration interface Ethernet 0
+# !
+# interface Ethernet0
+#  mtu 9100
+#  speed 25000
+#  unreliable-los auto
+#  no shutdown
+#  no dcbx enable
+#  no dcbx tlv-select pfc
+#  no dcbx tlv-select ets-conf
+#  no dcbx tlv-select ets-reco
+
+
+- name: Delete and set default DCBx Interface configurations
+  dellemc.enterprise_sonic.sonic_dcbx:
+    config:
+      interfaces:
+        - name: Ethernet0
+          enabled: false
+          pfc_tlv_enabled: false
+          ets_configuration_tlv_enabled: false
+          ets_recommendation_tlv_enabled: false
+    state: deleted
+
+# After State:
+# ------------
+#
+# sonic# show running-configuration interface Ethernet0
+# !
+# interface Ethernet0
+#  mtu 9100
+#  speed 25000
+#  unreliable-los auto
+#  no shutdown
+
+# Using "replaced" state for DCBx interface configuration
+#
+# Before State:
+# -------------
+#
+# sonic# show running-configuration interface Ethernet0
+# !
+# interface Ethernet0
+#  mtu 9100
+#  speed 25000
+#  unreliable-los auto
+#  no shutdown
+#  no dcbx tlv-select pfc
+#  no dcbx tlv-select ets-conf
+#  no dcbx tlv-select ets-reco
+#
+# sonic# show running-configuration interface Ethernet4
+# !
+# interface Ethernet4
+#  mtu 9100
+#  speed 25000
+#  unreliable-los auto
+#  no shutdown
+#  no dcbx enable
+#  no dcbx tlv-select pfc
+
+- name: Modify DCBx Interface configurations
+  dellemc.enterprise_sonic.sonic_dcbx:
+    config:
+      -interfaces:
+        - name: Ethernet0
+          enabled: false
+    state: replaced
+
+# After State:
+# ------------
+#
+# sonic# show running-configuration interface Ethernet0
```

Comment:
This example shows incorrect behavior.

Replacing the Ethernet0 configuration with only the "enabled" state specified in the configuration should change the other non-default values for the interface options to default values.

Reply:
corrected.